File: 17_05_2024_Find_Pair_Given_Difference.java
class Solution {
    public int findPair(int n, int x, int[] arr) {
        Arrays.sort(arr);
        int i = 0, j = 1;
        while (i < arr.length && j < arr.length) {
            if (arr[j] - arr[i] == x && j != i) {
                return 1;
            }
            if (arr[j] - arr[i] > x) {
                i++;
            } else {
                j++;
            }
        }
        return -1;
    }
}

File: 17_05_2024_Find_Pair_Given_Difference.cpp
class Solution {
  public:
    int findPair(int n, int x, vector<int> &arr) {
        unordered_set<int> elements;
        for (int i = 0; i < n; ++i) {
            if (elements.find(arr[i] + x) != elements.end() ||
                elements.find(arr[i] - x) != elements.end()) {
                return 1;
            }
            elements.insert(arr[i]);
        }
        return -1;
    }
};
